doc/examples/segmentation/plot_rolling_ball.py

```diff
@@ -0,0 +1,270 @@
+the entire image.
+
+Scikit-image implements this rolling-ball algorithm, as well as
+a generalized version which allows you to "roll" arbitrary ellipsoids. This
+generalized version is useful if you want to use different values for the
+spatial radius (``kernel_shape``) and the intensity amount
+(``intensity_vertex``).
+
+.. [1] Sternberg, Stanley R. "Biomedical image processing." Computer 1 (1983):
+    22-34. :DOI:`10.1109/MC.1983.1654163`
+
+
+Classic rolling ball
+-------------------------------
+
+In scikit-image, the rolling ball algorithm assumes that your background has
+low intensity (black), whereas the features have high intensity (white). If
+this is the case for your image, you can directly use the filter like:
+
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import time
+
+from skimage import (
+    data, morphology, util, io
+)
+
+
+def plot_result(image, background):
+    fig, ax = plt.subplots(nrows=1, ncols=3)
+
+    ax[0].imshow(image, cmap='gray')
+    ax[0].set_title('Original image')
+    ax[0].axis('off')
+
+    ax[1].imshow(background, cmap='gray')
+    ax[1].set_title('Background')
+    ax[1].axis('off')
+
+    ax[2].imshow(image - background, cmap='gray')
+    ax[2].set_title('Result')
+    ax[2].axis('off')
+
+    fig.tight_layout()
+
+
+image = data.coins()
+
+background = morphology.rolling_ball(image, radius=70.5)
+
+plot_result(image, background)
+plt.show()
+
+######################################################################
+# White background
+# ----------------
+#
+# If you have dark features on a bright background, you need to invert
+# the image before you pass it into the algorithm, and then invert the
+# result. This can be accomplished via:
+
+image = data.page()
+image_inverted = util.invert(image)
+
+background_inverted = morphology.rolling_ball(image_inverted, radius=45)
+filtered_image_inverted = image_inverted - background_inverted
+filtered_image = util.invert(filtered_image_inverted)
+background = util.invert(background_inverted)
+
+fig, ax = plt.subplots(nrows=1, ncols=3)
+
+ax[0].imshow(image, cmap='gray')
+ax[0].set_title('Original image')
+ax[0].axis('off')
+
+ax[1].imshow(background, cmap='gray')
+ax[1].set_title('Background')
+ax[1].axis('off')
+
+ax[2].imshow(filtered_image, cmap='gray')
+ax[2].set_title('Result')
+ax[2].axis('off')
+
+fig.tight_layout()
+
+plt.show()
+
+######################################################################
+# Be careful not to fall victim to an integer underflow when subtracting
+# a bright background. For example, this code looks correct, but may
+# suffer from an underflow leading to unwanted artifacts. You can see
+# this in the top right corner of the visualization.
+
+image = data.page()
+image_inverted = util.invert(image)
+
+background_inverted = morphology.rolling_ball(image_inverted, radius=45)
+background = util.invert(background_inverted)
+underflow_image = image - background  # integer underflow occurs here
```

Comment:
> isn't the background bounded by the image?

Yes for a black background, no for a white background (high intensity). For the latter, the background is an upper bound, not a lower bound, hence `image - background <= 0` on all points. For integer data things will underflow, creating the artifacts shown in the example.

(Built docs with pictures: https://351-2014929-gh.circle-artifacts.com/0/doc/build/html/auto_examples/segmentation/plot_rolling_ball.html#sphx-glr-auto-examples-segmentation-plot-rolling-ball-py)

Reply:
> `image - background <= 0` on all points

So why is there underflow only on a corner of the image...?

Also, don't you think the formulation above (`image_inverted - background_inverted`) is more intuitive, ie always subtracting the computed background from the input image, then fixing for "polarity"?

Reply:
It's not only in corner of the image, but the background intensity in that area causes funny results. Inversion + subtraction with `uint` types caught me off guard as well, which is why I added this demonstration to the examples. 

The underflow occurs for every pixel except for those where the background intensity exactly matches the intensity of the image. For those pixels, the intensity will be `0` as opposed to `256` which can't be represented. As a result `image - background` removes the background AND inverts the image except for pixels where background and image intensity match. These will be `0` causing black spots that should really be 255. A "hacked" fix ala `img[img==0]=255`, won't work here either, because 0 background pixels are now aliased with pixels that *should* have value 0 (intense features).

> Also, don't you think the formulation above (image_inverted - background_inverted) is more intuitive

I agree. This is why I do this in the example above, but point out this alternative in the "gotcha" section. The idea of not duplicating that is to show how one could explicitly handle this underflow ... maybe I'm making this too complicated?